lib/routes/routeBackbeat.js

```diff
@@ -1289,7 +1289,10 @@ function routeBackbeat(clientIP, request, response, log) {
                     [request.query.operation](request, response, log, next);
             }
             const versioningConfig = bucketInfo.getVersioningConfiguration();
-            if (!versioningConfig || versioningConfig.Status !== 'Enabled') {
+            // The following makes sure that only replication destination-related operations (non-GET requests)
+            // target buckets with versioning enabled.
+            // This allows lifecycle operations (which use GET requests) to work on non-versioned buckets.
+            if (request.method !== 'GET' && (!versioningConfig || versioningConfig.Status !== 'Enabled')) {
```

Comment:
lifecycle does not only perform `GET` operations : it also deletes objects (expiration), deletes data associated with objects (transition and restored-object-expiration), and puts metadata of objects...

so i fear such broad filtering in the router may be overkill.

→ Is there a way instead to identify the few "actions" performed by CRR replication (e.g. putObjectVersion, putObjectMetadata), maybe with some additional condition on the parameters/fields, to confirm? Then the check cna be performed direclty in each of these action handlers.
→ Otherwise, may be better to "isolate" the calls with specific CRR/lifecycle routes, or add an extra parameter to enforce the check (`IfVersioned` maybe ?)

→ Alternatively, instead of making it a "core" feature in the code, could it be achieve through policies? e.g. the replication user will not have permissions to write to non-versioned bucket? (need to check if this is a kind of policy we support or even can define, but would be an elegant and simple solution I think)

Reply:
- S3C Lifecycle expiration (transition is not supported yet) only uses backbeat client GET operation: `getMetadata`. Delete object/version is done by the S3 client itself that uses the s3 route.

> Is there a way instead to identify the few "actions" performed by CRR replication (e.g. putObjectVersion, putObjectMetadata)
- Replication uses `putData`, `putMetadata` and `batchDelete` 's backebatAPI routes. Limiting the versioning check to those actions makes sense and is more explicit. However `putMetadata` is also used by lifecycle transition so it will not solve the lifecycle transition future implementation.

- This PR is a fix, so I am not comfortable introducing new CRR/lifeycle routes for it. But it will be needed once we introduce lifecycle transition in S3C.

> the replication user will not have permissions to write to non-versioned bucket?
- I thought about it, but it does not exist.

Reply:
`batchDelete` route is used for lifecycle as well.

> This PR is a fix, so I am not comfortable introducing new CRR/lifeycle routes for it. But it will be needed once we introduce lifecycle transition in S3C.

Fine with me ; however the code will waterflow and in the not-so-long future we should converge Cloudserver: so whatever you change will break lifecycle in Zenko.

So I'd rather make the check conditional anyway, and let the "caller" decide if the bucket must be versionned: e..g. simply add an extra parameter to enforce the check (`IfVersioned` maybe ?) : then backbeat can be updated to required versioning if and if required, without introducing backbeat logic/constraints in cloudserver.

Reply:
Yes, I am aligned with it.